### PR TITLE
feat(whole project): implement floating footer with share buttons

### PR DESCRIPTION
**Pull request by Ugmaxie**

### CODE REVIEW

File: app/common/footer-floating/footer-floating.component.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            floatFooterContainer.classList.add('show-float-footer');
+
+            if (!scrollTop) {
+              floatFooterContainer.classList.remove('show-float-footer');
+            }
+          }
+        });
+      });
+
+    this.footerServiceSubscribe = this.footerService.getFooter()
```
Comment (Betrozov) [inline app/common/footer-floating/footer-floating.component.ts]: @Ugmaxie Why do you need this service?

File: app/common/footer-floating/footer-floating.component.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+  public ngOnInit():any {
+    this.scrollSubscribe = fromEvent(document, 'scroll')
+      .subscribe(() => {
+        let scrollTop = document.body.scrollTop || document.documentElement.scrollTop;
+
+        this.zone.run(() => {
+          let floatFooterContainer = this.element.querySelector('.floatfooter') as HTMLElement;
+
+          if (floatFooterContainer) {
```
Comment (Betrozov) [inline app/common/footer-floating/footer-floating.component.ts]: @Ugmaxie Why do you need this condition?

File: app/common/footer-floating/footer-floating.component.ts (hunk 1 above)
Comment (Ugmaxie) [inline app/common/footer-floating/footer-floating.component.ts]: That's phantom. :) We don't need this. Right and thanks.

File: app/common/footer-floating/footer-floating.component.ts (hunk 2 above)
Comment (Ugmaxie) [inline app/common/footer-floating/footer-floating.component.ts]: To be sure that element exists. But you're right.

File: app/common/footer-floating/footer-floating.css [hunk 3]
```diff
@@ -0,0 +1,71 @@
+
+.floatfooter {
+  overflow: hidden;
+  padding: 0 24px 0;
```
Comment (Betrozov) [inline app/common/footer-floating/footer-floating.css]: @Ugmaxie `padding: 0 22px;`

**Pull request merged**